io-engine/tests/snapshot_rebuild.rs

```diff
@@ -0,0 +1,227 @@
+use nix::errno::Errno;
+use once_cell::sync::OnceCell;
+use std::time::Duration;
+
+use io_engine::{
+    bdev::{device_create, device_destroy},
+    core::MayastorCliArgs,
+    rebuild::RebuildState,
+};
+
+pub mod common;
+use common::compose::MayastorTest;
+use io_engine::{
+    core::{LogicalVolume, ReadOptions, Share, ToErrno},
+    lvs::{Lvol, Lvs, LvsLvol},
+    pool_backend::PoolArgs,
+    rebuild::{RebuildJobOptions, SnapshotRebuildJob},
+    sleep::mayastor_sleep,
+};
+
+static MAYASTOR: OnceCell<MayastorTest> = OnceCell::new();
+
+fn get_ms() -> &'static MayastorTest<'static> {
+    MAYASTOR.get_or_init(|| {
+        MayastorTest::new(MayastorCliArgs {
+            ..Default::default()
+        })
+    })
+}
+
+async fn create_pool() -> Result<Lvs, String> {
```

Comment:
Can we not reuse these pool and replica create/destroy calls from other test modules?

Reply:
I did not find any existing helper functions for this.
There's a builder for pools/replicas/nexuses but it operates over gRPC only.
I guess I could modify that one to also work on the local reactor?

Reply:
> I did not find any existing helper functions for this. There's a builder for pools/replicas/nexuses but it operates over gRPC only. I guess I could modify that one to also work on the local reactor?

That'd be cool bcz it's a pain at times to juggle between tests running local reactor vs gRPC. But that can be done as a separate work item.

Reply:
Ticket raised, I'll work on this next. We also need to abstract it to work with non-LVS, ie: LVM :)